### PR TITLE
science bridge code 

### PR DESCRIPTION
**Pull request by cgiger00**
working as of mock comp

### CODE REVIEW

**Review by raytitan — COMMENTED** [review]
Found a few things

Overall I know the team has been having problems with receiving UART - but it would make both firmware and bridge code easier if the system used variable byte length packets. One of the best features of NMEA (which we copied) was variable length packets because of the \r\n at the end of each packet. Then you wouldn't need the messy padding (1111111) that is going on here.

File: jetson/science_bridge/src/__main__.py [hunk 1]
```diff
@@ -4,21 +4,23 @@
 '''
 import serial
 import asyncio
-# import Adafruit_BBIO.UART as UART
+import Adafruit_BBIO.UART as UART
```
[the remark this answers is not in the document]
Comment (cgiger00) [reply jetson/science_bridge/src/__main__.py:7]: it's used when we test on a beaglebone -- we last verified it on the BB which ig is why it's uncommented -- will fix

File: jetson/science_bridge/src/__main__.py [hunk 2]
```diff
@@ -4,21 +4,23 @@
...
 import numpy as np
-import re
+# import re
```
[the remark this answers is not in the document]
Comment (cgiger00) [reply jetson/science_bridge/src/__main__.py]: was used for parsing weird uart stuff - removed

File: jetson/science_bridge/src/__main__.py [hunk 3]
```diff
@@ -4,21 +4,23 @@
...
 import time
 from rover_common.aiohelper import run_coroutines
 from rover_common import aiolcm
 from rover_msgs import ThermistorData, MosfetCmd, RepeaterDrop, SpectralData, NavStatus, AmmoniaCmd
+
+
 class ScienceBridge():
     def __init__(self):
-        # UART.setup("UART4")  #  Specific to beaglebone
+        UART.setup("UART4")  # Specific to beaglebone
```
[the remark this answers is not in the document]
Comment (cgiger00) [reply jetson/science_bridge/src/__main__.py]: for remote testing we use it

**Comment by cgiger00** [comment]
For some reason when the packets are varied going from the jetson/beaglebone to the nucleo the HAL drivers don't receive the message correctly, we have one handler for all UART msgs and sometimes messages vary lengths like the mosfet one is a byte longer if the device is double digits, or the mosfet/ammonia one are different lengths bc of the information they carry. On the beaglebone/jetson receiving end though we are able to make it work with variable length @raytitan

**Review by jjtom34 — APPROVED** [review]

**Review by nkr101 — APPROVED** [review]

**Review by raytitan — APPROVED** [review]
Except for general comment about variable length packets, this looks fine overall

**Comment by cgiger00** [comment]
oops need to reopen

**Pull request merged**